### ISSUE TITLE
TLS connection to smtp server with password

### ISSUE THREAD

**Issue by samueleaton** [raydf/smtp.cr#2]
I'm trying to get something like you would see [here](https://www.digitalocean.com/community/tutorials/how-to-use-google-s-smtp-server). 

For reference, I've used a node.js module, [Nodemailer](https://github.com/nodemailer/nodemailer), that is able to implement that as seen in [this example](https://github.com/nodemailer/nodemailer/blob/master/examples/tl_dr.js). It looks like the bulk of the work happens in [this module](https://github.com/nodemailer/smtp-connection).

I would like to add some more secure functionality to smtp. Let me know if there is anything I can do to help.

**Comment by ray-delossantos** [comment]
Hi @samueleaton:

@sdogruyol asked for the same functionality for gmail smtp. There's 2 main features to implement, TLS connections (OpenSSL::SSL::Context::Client, https://crystal-lang.org/api/master/OpenSSL/SSL/Context/Client.html ) and Plain Auth (Super simple to implement). if you know the RFC's related to this features, you could help us with a Pull Request. I have a rough estimate to implement all the features in the first week of November, but i'm very bad with release dates :), so if you can help i'll appreciate it a lot.

**Comment by samueleaton** [comment]
Yeah looking at how some other libs do it it didn't seem like too much work. We should think of a good api and then get crackin'. 

Even though there is more demand for gmail, I think most would agree that it should work for many services and let the user be able to "plug 'n play". The nodemailer team did some work for us and has gathered a [list of common services](https://github.com/nodemailer/nodemailer-wellknown/blob/master/services.json) with domain alias, ports and security types.

**Comment by ray-delossantos** [comment]
Maybe take in consideration this api:

http://ruby-doc.org/stdlib-2.0.0/libdoc/net/smtp/rdoc/Net/SMTP.html

**Comment by samueleaton** [comment]
yeah i was reading the source code earlier today and saw that ruby was using the openssl lib like you suggested.

**Comment by ray-delossantos** [comment]
See also: https://github.com/crystal-lang/crystal/blob/36b2fc4a677721ef5c7f2a27d6270f72b2999019/src/http/client.cr#L154
here's the code of TLS connection usage.

**Comment by crisward** [comment]
I'm using https://github.com/arcage/crystal-email to auth with tls. Seems to work. Perhaps bringing the libs together somehow could help?

**Comment by ray-delossantos** [comment]
Hello @crisward 

Nice to know about this lib, i'll put a message in the readme file. Is there something in smtp.cr that you'll like to see in the crystal-email code base to cover in a PR?.

**Comment by crisward** [comment]
Crystal email doesn't support HTML emails, and smtp.cr doesn't support attachments or tls. 
It'd be nice if we had one library with all the features.
